apps/judicial-system/backend/src/app/modules/case/internalCase.service.ts

```diff
@@ -1241,17 +1247,39 @@ export class InternalCaseService {
       where: {
         type: CaseType.INDICTMENT,
         id: caseId,
-        // The national id could be without a hyphen or with a hyphen so we need to
-        // search for both
-        '$defendants.national_id$': normalizeAndFormatNationalId(nationalId),
+        state: { [Op.not]: CaseState.DELETED },
+        isArchived: false,
+        // This select only defendants with the given national id, other defendants are not included
+        '$defendants.national_id$':
+          normalizeAndFormatNationalId(defendantNationalId),
       },
     })
 
-    if (!caseById) {
-      throw new NotFoundException(`Case ${caseId} not found`)
+    if (!theCase) {
+      throw new NotFoundException(`Case ${caseId} does not exist`)
+    }
+
+    return theCase
+  }
+
+  async getDefendantIndictmentCase(
+    theCase: Case,
+    defendantNationalId: string,
+  ): Promise<Case> {
+    const subpoena = theCase.defendants?.[0].subpoenas?.[0]
+
+    if (!subpoena) {
+      return theCase
+    }
+
+    const latestSubpoena = await this.subpoenaService.getSubpoena(subpoena)
```

Comment:
The subpoena service takes care of refreshing the subpoena info from the police system as needed.